src/components/App.js

```diff
@@ -29,25 +29,22 @@ export class App extends Component {
     }
   }
 
-  getFavorites = async () => {
-    const { currentUser } = this.props
-    if(currentUser === null) {
-      return
-    } else {
-      const favoriteMovies = await fetchData(`http://localhost:3001/api/v1/users/${currentUser.id}/moviefavorites`)
-      console.log('in getFavorites--->>>', favoriteMovies)
-      return favoriteMovies
-    }
-  }
-
-  makeFavorite = async (movieInfo, id) => {
-    const { currentUser, saveFavorited } = this.props
+  toggleFavorite = async (movieInfo, id) => {
+    const { currentUser, saveNewFavorite, deleteStoredFavorite, favorited } = this.props
     if(currentUser === null) {
       return
     } else {
-      const postedFavorite = await postFavorite(movieInfo, id)
-      saveFavorited(movieInfo)
-      return postedFavorite
+      if(favorited.find(favoriteMovie => {
```

Comment:
We will probably want to break this up a bit to make it more dry

Reply:
definitely! your help is requested on that